server/sonar-server/src/main/java/org/sonar/server/user/ServerUserSession.java

```diff
@@ -48,28 +50,24 @@
   @CheckForNull
   private final UserDto userDto;
   private final DbClient dbClient;
-  private final Supplier<List<GroupDto>> groups;
+  private final OrganizationFeature organizationFeature;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+  private final Supplier<List<GroupDto>> groups = Suppliers.memoize(this::loadGroups);
+  private final Supplier<Boolean> isSystemAdministratorSupplier = Suppliers.memoize(this::loadIsSystemAdministrator);
   private SetMultimap<String, String> projectUuidByPermission = HashMultimap.create();
   private SetMultimap<String, String> permissionsByOrganizationUuid;
   private Map<String, String> projectUuidByComponentUuid = newHashMap();
   private List<String> projectPermissionsCheckedByUuid = new ArrayList<>();
 
-  private ServerUserSession(DbClient dbClient, @Nullable UserDto userDto) {
-    this.userDto = userDto;
+  ServerUserSession(DbClient dbClient, OrganizationFeature organizationFeature,
+    DefaultOrganizationProvider defaultOrganizationProvider, @Nullable UserDto userDto) {
     this.dbClient = dbClient;
-    this.groups = Suppliers.memoize(this::loadGroups);
-  }
-
-  public static ServerUserSession createForUser(DbClient dbClient, UserDto userDto) {
-    requireNonNull(userDto, "UserDto must not be null");
-    return new ServerUserSession(dbClient, userDto);
-  }
-
-  public static ServerUserSession createForAnonymous(DbClient dbClient) {
-    return new ServerUserSession(dbClient, null);
+    this.organizationFeature = organizationFeature;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+    this.userDto = userDto;
   }
 
-  private List<GroupDto> loadGroups() {
+    private List<GroupDto> loadGroups() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)